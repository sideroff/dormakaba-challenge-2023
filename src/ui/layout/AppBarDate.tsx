import { useState, useEffect } from 'react';
import Typography from '@mui/material/Typography';
import { getLocaleString, DateTime } from '@/lib/dateTime';

export function AppBarDate() {
  const [currentDate, setCurrentDate] = useState<string | undefined>();

  /**
   * make sure that current date calculation is only executed on client side
   */
  useEffect(() => {
    setCurrentDate(getLocaleString(new Date(), DateTime.DATE_SHORT));
  }, []);

  return (
    <Typography sx={{ ml: 2, color: 'common.black' }}>{currentDate}</Typography>
  );
}
