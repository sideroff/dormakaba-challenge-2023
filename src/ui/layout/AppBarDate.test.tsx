import { render } from '@testing-library/react';
import { AppBarDate } from './AppBarDate';

describe('AppBarDate', () => {
  beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2023-02-12T14:30:20.124'));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render correctly', () => {
    const { container } = render(<AppBarDate />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
