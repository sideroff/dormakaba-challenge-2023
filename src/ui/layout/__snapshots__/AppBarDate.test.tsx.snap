// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarDate should render correctly 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-p1x2na-MuiTypography-root"
>
  2/12/2023
</p>
`;
